From e86192c31529e6c9f8c3d6acbae57d56c26f79c0 Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Thu, 2 Jul 2020 22:32:13 +0100
Subject: [PATCH 12/18] FindWestonClient.cmake: Update to use weston-desktop
 pkgconfig

The symbols which we need are in libweston-desktop library.
So update from weston.pc to libweston-desktop.pc so we pick
up the correct linker flags.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
---
 cmake/modules/FindWestonClient.cmake | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/cmake/modules/FindWestonClient.cmake b/cmake/modules/FindWestonClient.cmake
index f88b6f7..295b27c 100644
--- a/cmake/modules/FindWestonClient.cmake
+++ b/cmake/modules/FindWestonClient.cmake
@@ -31,27 +31,24 @@
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 find_package(PkgConfig)
-pkg_check_modules(PC_WESTON weston)
+pkg_check_modules(PC_WESTON libweston-desktop-8)
 
-find_library(WESTON_CLIENT_LIB NAMES weston-desktop-5 weston-desktop-6
+find_library(WESTON_CLIENT_LIB NAMES weston-desktop-8
         HINTS ${PC_WESTON_LIBDIR} ${PC_WESTON_LIBRARY_DIRS}
 )
 
-set (WESTON_CLIENT_LIBRARIES ${PC_WESTON_LIBRARIES})
-
 if(PC_WESTON_FOUND AND NOT TARGET WestonClient::WestonClient)
-    set(WESTON_CLIENT_LIB_CLIENT_LINK_LIBRARIES "${WESTON_CLIENT_LIB}")
     add_library(WestonClient::WestonClient UNKNOWN IMPORTED)
     set_target_properties(WestonClient::WestonClient PROPERTIES
             IMPORTED_LINK_INTERFACE_LANGUAGES "C"
             IMPORTED_LOCATION "${WESTON_CLIENT_LIB}"
             INTERFACE_INCLUDE_DIRECTORIES ""
-            INTERFACE_COMPILE_OPTIONS "${WESTON_CLIENT_CFLAGS_OTHER}"
-            INTERFACE_LINK_LIBRARIES "${WESTON_CLIENT_LIB_CLIENT_LINK_LIBRARIES}"
+            INTERFACE_COMPILE_OPTIONS "${PC_WESTON_DEFINITIONS}"
+            INTERFACE_LINK_LIBRARIES "${PC_WESTON_LIBRARIES}"
             )
 endif()
 
 include(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(PC_WESTON DEFAULT_MSG PC_WESTON_FOUND)
 
-mark_as_advanced(WESTON_CLIENT_LIBRARIES)
+mark_as_advanced(PC_WESTON_LIBRARIES)
-- 
2.17.1

